acceptance/tests/direct_puppet/static_catalog_env_control.rb

```diff
@@ -0,0 +1,348 @@
+}
+profiler: {
+}
+versioned-code : {
+  code-id-command : #{@coderoot}/code_id.sh
+  code-content-command : #{@coderoot}/code_content.sh
+}
+",
+}
+
+file { '#{@coderoot}/code_id.sh' :
+  ensure => file,
+  mode => "0755",
+  content => '#! /bin/sh
+echo "code_version_1"
+',
+}
+
+file { '#{@coderoot}/code_content.sh' :
+  ensure => file,
+  mode => "0755",
+  content => '#! /bin/sh
+if [ $2 == "code_version_1" ] ; then
+  echo "code_version_1"
+else
+  echo "newer_version"
+fi
+',
+}
+  
+file { '#{@coderoot}/environments/production/environment.conf':
+  ensure => file,
+  mode => "0644",
+  content => 'environment_timeout = 0
+',
+}
+  
+file { '#{@coderoot}/environments/canary/environment.conf':
+  ensure => file,
+  mode => "0644",
+  content => 'environment_timeout = 0
+static_catalogs = false
+',
+}
+
+file { '#{@coderoot}/environments/production/manifests/site.pp':
+  ensure => file,
+  mode => "0644",
+  content => "node default {
+  include hello
+}
+",
+}
+
+file { '#{@coderoot}/environments/canary/manifests/site.pp':
+  ensure => file,
+  mode => "0644",
+  content => "node default {
+  include can_hello
+}
+",
+}
+
+file { '#{@coderoot}/environments/production/modules/hello/files/hello_msg':
+  ensure => file,
+  mode => "0644",
+  content => "Hello message from production/hello module, content from source attribute.
+",
+}
+
+file { '#{@coderoot}/environments/canary/modules/can_hello/files/hello_msg':
+  ensure => file,
+  mode => "0644",
+  content => "Hello message from canary/can_hello module, content from source attribute.
+",
+}
+MANIFEST
+
+step 'apply main manifest, static_catalogs unspecified in global scope, unspecified in production environment, disabled in canary environment'
+
+apply_manifest_on(master, @manifest, :catch_failures => true)
+
+step 'start puppet server'
+with_puppet_running_on master, @master_opts, @coderoot do
+  agents.each do |agent|
+
+    resdir = @agent_result_dirs[agent.hostname]
+    apply_manifest_on(master, <<MANIFESTAGENT, :catch_failures => true)
+file { '#{@coderoot}/environments/production/modules/hello/manifests/init.pp':
+  ensure => file,
+  mode => "0644",
+  content => "class hello {
+  notice('hello from production-hello')
+  file { '#{resdir}' :
+    ensure => directory,
+    mode => '0755',
+  }
+  file { '#{resdir}/prod_hello_from_puppet_uri' :
+    ensure => file,
+    mode => '0644',
+    source => 'puppet:///modules/hello/hello_msg',
+  }
+}",
+}
+
+file { '#{@coderoot}/environments/canary/modules/can_hello/manifests/init.pp':
+  ensure => file,
+  mode => "0644",
+  content => 'class can_hello {
+  notice("hello from production-hello")
+  file { "#{resdir}":
+    ensure => directory,
+    mode => "0755",
+  }
+  file { "#{resdir}/can_hello_from_puppet_uri" :
+    ensure => file,
+    mode => "0644",
+    source => "puppet:///modules/can_hello/hello_msg",
+  }
+}',
+}
+MANIFESTAGENT
+
+  step "Add versioned-code parameters to puppetserver.conf and ensure the server is running" 
```

Comment:
And this section modifying tk_config should be moved outside the agents loop. The config should be the same for all agents.